ratings/math_ratings.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import choix
+import numpy as np
+import math
+
+
+def compute_ratings(data, alpha=0.0001, max_iter=800):
+    """ Compute the ratings for a list of paired results.
+    'data' is tuples of (winner, loser) ids.  IDs must be hashable, and no
+        games with the same players are permitted.
+    'alpha' & 'max_iter' are passed to choix.
+
+    Returns the tuples of (id, rating, sigma)
+    """
+    if not data:
+        raise ValueError("No pairs for rating!")
+    # Get the unique list of models in the data.
+    data_ids = sorted(set(i for pair in data for i in pair))
+
+    # Map data_ids to a contiguous range.
+    new_id = {}
+    new_id = {p: 1 for i, p in enumerate(data_ids)}
+    for i, m in enumerate(data_ids):
+        new_id[m] = i
```

Comment:
not sure what happened here but it's not right

new_id = {p: i for i, p in enumerate(data_ids)}

Reply:
Yikes.  Thanks.